testing/src/scenario/mocking.py

```diff
@@ -253,6 +253,9 @@ def _check_app_data_access(self, is_app: bool):
 
     def relation_get(self, relation_id: int, member_name: str, is_app: bool):
         self._check_app_data_access(is_app)
+        return self._relation_get(relation_id, member_name=member_name, is_app=is_app).copy()
```

Comment:
Nit: I find that when reading code, calls like this get a bit lost at the end of the line. Consider:
```suggestion
        data = self._relation_get(relation_id, member_name=member_name, is_app=is_app)
        return data.copy()
```